app/src/apdu_handler.c

```diff
@@ -123,7 +124,14 @@ __Z_INLINE void handleGetAddr(volatile uint32_t *flags, volatile uint32_t *tx, u
         THROW(APDU_CODE_DATA_INVALID);
     }
     if (requireConfirmation) {
-        view_review_init(addr_getItem, addr_getNumItems, app_reply_address);
+        switch (kind) {
+            case addr_ed25519:
+                view_review_init(addr_getItem_ed25519, addr_getNumItems, app_reply_address);
+                break;
+            case addr_secp256k1:
+                view_review_init(addr_getItem_secp256k1, addr_getNumItems, app_reply_address);
+                break;
+        }
```

Comment:
probably it handled in app_fill_address but can we add a default with error here?